src/PlatformEngines.jl

```diff
@@ -214,31 +214,46 @@ function probe_platform_engines!(;verbose::Bool = false)
     # package_opts_functor, list_opts_functor, parse_functor).  The probulator
     # will check each of them by attempting to run `$test_cmd`, and if that
     # works, will set the global compression functions appropriately.
-    gen_7z = (p) -> (unpack_7z(p), package_7z(p), list_7z(p), parse_7z_list)
+    gen_7z = (p) -> (unpack_7z(p), package_7z(p), list_7z(p), parse_7z_list, r"Path = ([^\r\n]+)\r?\n(?:[^\r\n]+\r?\n)+Symbolic Link = ([^\r\n]+)"s)
     compression_engines = Tuple[]
 
+    write("demoTar.txt","Demo file for tar listing")
     for tar_cmd in [`tar`, `busybox tar`]
+        # try to determine the tar list format
+        local symlink_parser
+        try
+            tarListing = read(pipeline(`$tar_cmd -c demoTar.txt`,`$tar_cmd -tv`), String)
+            m = match(r"((?:\S+\s+)+?)demoTar\.txt", tarListing)[1]
+            nargs = length(split(m, " "; keepempty = false))
+            symlink_parser = Regex("^l(?:\\S+\\s+){$nargs}(.+?)(?: -> (.+?))?\\r?\$", "m")
+        catch
+            # generic expression for symlink parsing
+            # this will fail, if the symlink contains space characters (which is highly improbable, though)
+            symlink_parser = r"^l.+? (\S+?)(?: -> (.+?))?\r?$"m
+        end
         # Some tar's aren't smart enough to auto-guess decompression method. :(
-        unpack_tar = (tarball_path, out_path) -> begin
+        unpack_tar = (tarball_path, out_path, excludelist = "") -> begin
             Jjz = "z"
             if endswith(tarball_path, ".xz")
                 Jjz = "J"
             elseif endswith(tarball_path, ".bz2")
                 Jjz = "j"
             end
-            return `$tar_cmd -x$(Jjz)f $(tarball_path) --directory=$(out_path)`
+            return `$tar_cmd -x$(Jjz)f $(tarball_path) --directory=$(out_path) $(excludelist=="" ? [] : ["--exclude-from=" * excludelist])`
         end
         package_tar = (in_path, tarball_path) ->
             `$tar_cmd -czvf $tarball_path -C $(in_path) .`
-        list_tar = (in_path) -> `$tar_cmd -tzf $in_path`
+        list_tar = (in_path; verbose = false) -> `$tar_cmd $(verbose ? "-tzvf" : "-tzf") $in_path`
```

Comment:
Suggested change:

`$tar_cmd -tz$(verbose ? "v" : "")f`

Reply:
that was my first version, but it didn't work, because empty strings and strings containing blanks are added with quotation marks. Meanwhile, I found out that you have to use backticks :-)

Reply:
string interpolation in commands makes some very strange things, e.g.
```
julia> (x->`tar -tc$(x)f me`).(a)
4-element Array{Cmd,1}:
 `tar -tcvf me`
 `tar -tcvf me`
 `tar me`
 `tar -tcf me`

julia> (x->`tar $(x) me`).(a)
4-element Array{Cmd,1}:
 `tar v me`
 `tar v me`
 `tar me`
 `tar '' me`
```
I think, this is not intentional, is it?

Reply:
I don't think `Cmd`'s are made to be interpolated into other `Cmd`s; try only using `String`s:

```julia
julia> (v -> `tar -tz$(v ? "v" : "")f me`)(true)
`tar -tzvf me`

julia> (v -> `tar -tz$(v ? "v" : "")f me`)(false)
`tar -tzf me`
```